### DISCUSSION TITLE
How to create a new File Storage provider?

### DISCUSSION THREAD

**Discussion by abdonrd** [directus/directus#14393] [Archive]
In the documentation:

> By default, Directus stores all uploaded files locally on disk. However, you can also configure Directus to use S3, Google Cloud Storage, or Azure. 

https://docs.directus.io/self-hosted/config-options/#file-storage

But how we can create our own provider?
We want to use the [IBM Cloud Object Storage](https://www.ibm.com/cloud/object-storage) ([IBM/ibm-cos-sdk-js](https://github.com/IBM/ibm-cos-sdk-js)).

Thanks!

**Comment by rijkvanzanten** [comment]
It's currently not an extension type yet. If you wanna give it a shot, you could take a look at the `@directus/drive-s3` package and see if you're comfortable in making a copy and updating it for ibm-cos-sdk-js 🙂

**Reply by abdonrd** [reply]
Oh! Looks like ibm-cos-sdk-js it's an aws-sdk, and we can use the S3 directly! 🎉
